### PR TITLE
GAPI-31611 [Agent API Simulator]Support of the Azure platform

### PR DESCRIPTION
**Pull request by VadymKondra**
Created mechanism to download assets, if the files lays on different hosts

### CODE REVIEW

File: src/scripts/downloadUiAssets.js [hunk 1]
```diff
@@ -322,12 +388,32 @@ async function main() {
...
   try {
-    // Check versions compatibility
-    await exports.checkCompatibility(await getVersions(url), url);
-    // Get Workspace Web Edition archive
-    await getArchive(url + 'ui/wwe/', './ui-assets/wwe');
-    // Get GWS Auth UI archive
-    await getArchive(url + 'ui/auth/', './ui-assets/auth');
+    let isAzurePlatform;   
+    isAzurePlatform = await checkUrls(url);
+    console.log('result profile', isAzurePlatform);
```
Comment (sblecon) [inline src/scripts/downloadUiAssets.js:393]: Do you plan to expose this trace to customer?

File: src/scripts/downloadUiAssets.js [hunk 2]
```diff
@@ -322,12 +388,32 @@ async function main() {
...
+    if(isAzurePlatform === 'retry') {
+      isAzurePlatform = await checkUrls(url, 'ui/wwe/');
+      console.log('result profile 2', isAzurePlatform);
```
Comment (sblecon) [inline src/scripts/downloadUiAssets.js:396]: Do you plan to expose this trace to customer?

File: src/scripts/downloadUiAssets.js [hunk 3]
```diff
@@ -183,6 +205,48 @@ function getVersion(url) {
...
+      let profile = body;
+      resolve((profile && profile.includes('window.genesys.wwe.env.GWS_SERVICE_URL')));
+    })
+  }));
+}
+
+function getAuthURL(url) {
+  return (new Promise((resolve, reject) => {
+    console.log("Getting auth url from: ", url)
+    const redirect = url.replace('wwe-','gws.api01-');
```
Comment (sblecon) [inline src/scripts/downloadUiAssets.js:229]: Why you don't use information from profile.js ?
Based on such kind of lines:
`window.genesys.wwe.env.EXPRESSION_WWE_URL_CAPTURING_GROUPS='https:\/\/wwe-([A-Za-z0-9]+)\.prod([0-9]+)\.genesysengage\.com\/.*';
window.genesys.wwe.env.GWS_SERVICE_URL='https://gws.api01-$WWE_URL_GROUP_1$.prod$WWE_URL_GROUP_2$.genesysengage.com';

window.genesys.wwe.env.EXPRESSION_WWE_URL_CAPTURING_GROUPS_1='https:\/\/(.*)-wwe-(.*)-pvt\.genesysengage\.com\/.*';
window.genesys.wwe.env.GWS_SERVICE_URL_1='https://$WWE_URL_GROUP_1$-gapi-$WWE_URL_GROUP_2$-pvt.genesysengage.com';`

**Review by sblecon — CHANGES_REQUESTED** [review]

File: src/scripts/downloadUiAssets.js (hunk 1 above)
Comment (VadymKondra) [reply src/scripts/downloadUiAssets.js]: Deleted this console.log

File: src/scripts/downloadUiAssets.js (hunk 3 above)
Comment (VadymKondra) [reply src/scripts/downloadUiAssets.js]: Fixed algorithm

File: src/scripts/downloadUiAssets.js [hunk 4]
```diff
@@ -322,12 +403,28 @@ async function main() {
...
-    await getArchive(url + 'ui/wwe/', './ui-assets/wwe');
-    // Get GWS Auth UI archive
-    await getArchive(url + 'ui/auth/', './ui-assets/auth');
+    let isAzurePlatform;   
+    isAzurePlatform = await checkUrls(url);
+    if(isAzurePlatform === 'retry') {
+      isAzurePlatform = await checkUrls(url, 'ui/wwe/');
+    }
+    if (isAzurePlatform === true) {
+      authUrl = await getGwsApiUri(url)+'/ui/auth/';
```
Comment (sblecon) [inline src/scripts/downloadUiAssets.js:412]: Isn't guaranty that in the future Auth UI will be  on the same domain than GWS.
Why did you not try to get the information from GWS like mentioned in the Jira ticket?

**Review by sblecon — CHANGES_REQUESTED** [review]

File: src/scripts/downloadUiAssets.js [hunk 5]
```diff
@@ -322,12 +403,28 @@ async function main() {
...
   try {
-    // Check versions compatibility
-    await exports.checkCompatibility(await getVersions(url), url);
-    // Get Workspace Web Edition archive
-    await getArchive(url + 'ui/wwe/', './ui-assets/wwe');
-    // Get GWS Auth UI archive
-    await getArchive(url + 'ui/auth/', './ui-assets/auth');
+    let isAzurePlatform;   
+    isAzurePlatform = await checkUrls(url);
+    if(isAzurePlatform === 'retry') {
```
[the remark this answers is not in the document]
Comment (VadymKondra) [reply src/scripts/downloadUiAssets.js]: Fixed

File: src/scripts/downloadUiAssets.js (hunk 4 above)
Comment (VadymKondra) [reply src/scripts/downloadUiAssets.js]: Fixed

File: src/scripts/downloadUiAssets.js [hunk 6]
```diff
@@ -54,6 +54,28 @@ async function getVersions(baseUrl) {
   return (null);
 }
 
+async function getVersionsForAzure(wweUrl, authUrl) {
```
Comment (sblecon) [inline src/scripts/downloadUiAssets.js:57]: Should be renamed by getVersionsForSeparatedUIAndServicePlatform

**Review by sblecon — CHANGES_REQUESTED** [review]

File: src/scripts/downloadUiAssets.js (hunk 6 above)
Comment (VadymKondra) [reply src/scripts/downloadUiAssets.js]: Fixed

**Review by Karkhutvy — APPROVED** [review]

**Review by sblecon — APPROVED** [review]

**Pull request merged**